packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,232 @@
+import {
+  Attachment,
+  Bundle,
+  CodeableConcept,
+  Coding,
+  DocumentReference,
+  Extension,
+  Identifier,
+} from "@medplum/fhirtypes";
+import { executeWithNetworkRetries } from "@metriport/shared";
+import { buildDayjs } from "@metriport/shared/common/date";
+import { MedicalDataSource, isMedicalDataSource } from "..";
+import { createAttachmentUploadFileName } from "../../domain/document/upload";
+import {
+  CdaCodeCv,
+  CdaInstanceIdentifier,
+  CdaOriginalText,
+  ConcernActEntryAct,
+  EffectiveTimeLowHigh,
+} from "../../fhir-to-cda/cda-types/shared-types";
+import { executeAsynchronously } from "../../util/concurrency";
+import { Config } from "../../util/config";
+import { out } from "../../util/log";
+import { sizeInBytes } from "../../util/string";
+import { uuidv4 } from "../../util/uuid-v7";
+import { S3Utils, UploadParams } from "../aws/s3";
+import { cqExtension } from "../carequality/extension";
+import { cwExtension } from "../commonwell/extension";
+import { makeFhirApi } from "../fhir/api/api-factory";
+import { convertCollectionBundleToTransactionBundle } from "../fhir/bundle/convert-to-transaction-bundle";
+import { buildDocIdFhirExtension } from "../fhir/shared/extensions/doc-id-extension";
+
+const s3Utils = new S3Utils(Config.getAWSRegion());
```

Comment:
I prefer this pattern:
 
 ```
const region = Config.getAWSRegion();

function getS3UtilsInstance(): S3Utils {
  return new S3Utils(region);
}
```

over `const s3Utils = new S3Utils(Config.getAWSRegion());` as it's a bit too global for me but up to you.

Reply:
I added this. But if you look up this exact line:
```const s3Utils = new S3Utils(Config.getAWSRegion());```

you'll find like 10 more instances of this throughout the code base. But ya, never too late to start writing better code! 🫡